astropy/units/tests/test_equivalencies.py

```diff
@@ -627,6 +628,58 @@ def test_spectraldensity_not_allowed(from_unit, to_unit):
         to_unit.to(from_unit, 1, u.spectral_density(1 * u.AA))
 
 
+@dataclasses.dataclass(eq=False)
+class DuckQuantity(np.lib.mixins.NDArrayOperatorsMixin):
+
+    quantity: u.Quantity
+
+    def __array_ufunc__(self, function, method, *inputs, **kwargs):
+
+        inputs = [
+            inp.quantity if isinstance(inp, type(self)) else inp for inp in inputs
+        ]
+
+        result = getattr(function, method)(*inputs, **kwargs)
+        if result is not NotImplemented:
+            return type(self)(result)
+        return NotImplemented
+
+    def to(self, unit: u.Unit, equivalencies: list[tuple] = []) -> "DuckQuantity":
+        return type(self)(self.quantity.to(unit, equivalencies))
+
+    def to_value(
+        self, unit: None | u.Unit = None, equivalencies: list[tuple] = []
+    ) -> "DuckQuantity":
+        return type(self)(self.quantity.to_value(unit, equivalencies))
+
+
+@pytest.mark.parametrize(
+    argnames="energy",
+    argvalues=[
+        DuckQuantity(1 * u.erg),
+    ],
+)
+@pytest.mark.parametrize(
+    argnames="wavelength",
+    argvalues=[
+        DuckQuantity(300 * u.nm),
+    ],
+)
```

Comment:
```suggestion
@pytest.mark.parametrize(
    ("energy", "wavelength"),
    argvalues=[
        (DuckQuantity(1 * u.erg), DuckQuantity(300 * u.nm)),
    ],
)
```

Reply:
not sure how parametrization helps here if a single set of arguments is used, but if we plan to add more, then :+1: to @nstarman's suggestion/